lib/oxidized/model/awplus.rb

```diff
@@ -0,0 +1,51 @@
+class Awplus < Oxidized::Model
+
+  #Allied Telesis Alliedware Plus Model#
+
+  prompt /^(\r?[\w.@:\/-]+[#>]\s?)$/
+  comment  '! '
+
+  #This shows the system information at the top and comments it out#
+  cmd 'show system' do |cfg|
+    cfg = cfg.each_line.to_a[0...-2]
+
+
+   # Strip system (up)time/ram/free flash.
+    cfg = cfg.reject { |line| line.match /System Status\s*.*/ }
```

Comment:
could you join all these `#reject` call into one? this way we don't iterate several times over each line 😃

Reply:
you can just join them like this, since they always return the modified `Array`:

``` ruby
cfg = cfg
  .reject { |line| line.match /RAM\s*:.*/ }
  .reject { |line| line.match /Uptime\s*:.*/ }
  .reject { |line| line.match /Flash\s*:.*/ }
```

we could also use `#reject!` that changes the variable instead 🤓 

``` ruby
cmd 'some command' do |cfg|
  comment cfg
    .reject! { |line| line.match /RAM\s*:.*/ }
    .reject! { |line| line.match /Uptime\s*:.*/ }
    .reject! { |line| line.match /Flash\s*:.*/ }
end
```

Reply:
Not yet in the patch but I've done this to fix it:

```
  #Adds "Show system" output to start of config. 

  cmd 'Show System' do |cfg|
    comment cfg.insert(0,"--------------------------------------------------------------------------------! \n")
    #Unhash below to write a comment in the config file.
    cfg.insert(0,"Starting: Show system cmd \n")
    cfg << "\n\nEnding: show system cmd"
    comment cfg << "\n--------------------------------------------------------------------------------! \n\n"
    #Removes the following lines from "show system" in output file. This ensures oxidized diffs are meaningful. 
    comment cfg.each_line.reject { |line| 
                                    line.match /^$\n/ or #Remove blank lines in "sh sys"
                                    line.match /System Status\s*.*/ or
                                    line.match /RAM\s*:.*/ or 
                                    line.match /Uptime\s*:.*/ or 
                                    line.match /Flash\s*:.*/ or 
                                    line.match /Current software\s*:.*/ or 
                                    line.match /Software version\s*:.*/ or 
                                    line.match /Build date\s*:.*/ }.join
  end
```